### PR TITLE
Prefill input field with url parameter. Start the process automatically.

### PR DESCRIPTION
**Pull request by martinkraft**
Another try from a reseted version

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -114,3 +114,14 @@ $feedbackForm.submit( function( e ) {
...
 } );
+
+$(function() {
+	// Instant start if input field is filled
+	var $fileForm = $('#file-form'),
+		$fileFormInput = $fileForm.find('#file-form-input'),
+		fileFormVal = $fileFormInput.val();
+
+	if (fileFormVal && fileFormVal !== '') {
+		$fileForm.find('[type="submit"]').trigger('click');
```
Comment (ricordisamoa) [inline js/main.js]: Why not `$fileForm.submit()`?

File: index.php [hunk 2]
```diff
@@ -13,6 +13,10 @@
 // Get the base html to output
 $html = file_get_contents( __DIR__ . '/index.base.html' );
 $html = str_replace( '{{i18n.lang}}', $lang, $html );
+$html = str_replace( '{{inputattr}}',
+    isset( $_GET['url'] ) ? ' value="'.$_GET['url'].'"' : '',
```
Comment (ricordisamoa) [inline index.php]: Why not `htmlspecialchars()`?

File: js/main.js (hunk 1 above)
Comment (martinkraft) [inline js/main.js]: Because I wasn't sure, whether or not there are any other actions bound to this click-event.

**Comment by martinkraft** [comment]
@ricordisamoa I just updated both files based on your input.

**Comment by addshore** [comment]
The CI is still failing (3 code style errors) report @ https://travis-ci.org/wmde/Lizenzhinweisgenerator/jobs/207893046

```
2.17s$ ./node_modules/jscs/bin/jscs .
disallowSpaceAfterKeywords: Unexpected whitespace between if and ( at ./js/main.js :
   123 |  fileFormVal = $fileFormInput.val();
   124 |
   125 | if (fileFormVal && fileFormVal !== '') {
-----------^
   126 |  $fileForm.submit();
   127 | }
requireSpacesInsideParentheses: Missing space after opening round bracket at ./js/main.js :
   123 |  fileFormVal = $fileFormInput.val();
   124 |
   125 | if (fileFormVal && fileFormVal !== '') {
-------------^
   126 |  $fileForm.submit();
   127 | }
requireSpacesInsideParentheses: Missing space before closing round bracket at ./js/main.js :
   123 |  fileFormVal = $fileFormInput.val();
   124 |
   125 | if (fileFormVal && fileFormVal !== '') {
----------------------------------------------^
   126 |  $fileForm.submit();
   127 | }
3 code style errors found.
```

**Comment by martinkraft** [comment]
@addshore Wouldn't it be nice to include an according EditorConfig-file, if such a special code style is required? Because otherwise normal IDEs will screw up this requirements everytime they reformat a section of code.

**Comment by martinkraft** [comment]
I just saw, that there is a .jscrc in the project. But `"requireSpacesInsideBrackets": null` leaves me kind of puzzled...

**Comment by tobijat** [comment]
@jakobw @manicki @WMDE-Fisch if you find a few mins to have a look at this, that would be awesome! :)

**Comment by WMDE-Fisch** [comment]
Hi @martinkraft 
> I just saw, that there is a .jscrc in the project. But "requireSpacesInsideBrackets": null leaves me kind of puzzled...

The rule `requireSpacesInsideBrackets` relates to these square brackets `[]` the other brackets are called round brackets in the rule sets. ( my dictionary states, that in AE "bracket" always means "square bracket" )

> Wouldn't it be nice to include an according EditorConfig-file, if such a special code style is required?

Feel free and add a PR :->. - No, seriously I am not sure how to do this, but probably this would be a good idea for all our projects!

Related to CR, I am not entirely sure how we handle XSS risks in the app. Maybe Symfony takes care of that for the globals. I am not sure if `htmlspecialchars` is sufficient nowadays. Ping @jakobw @manicki

**Comment by ricordisamoa** [comment]
FWIW, without `htmlspecialchars` any HTML seems to get through, that's why I suggested it in the first place :wink:

**Comment by WMDE-Fisch** [comment]
> FWIW, without htmlspecialchars any HTML seems to get through, that's why I suggested it in the first place :wink:

And thats a very good thing! :-) I only wondered if we should do more.

**Comment by martinkraft** [comment]
Since almost the entire application is running on the client side, with no own database, without any own server logic or write processes, even a successful injection shouldn't cause any serious harm?!

P.S. @WMDE-Fisch: http://editorconfig.org/

**Comment by manicki** [comment]
I don't know why htmlencoding input wouldn't be enough, so +2 from my side. Thanks @martinkraft!

Regarding: 
> Wouldn't it be nice to include an according EditorConfig-file, if such a special code style is required?

I am going to add it in a bit. BTW, note that this "special code style" is quite close to what MediaWiki uses as its "standard" if I am completely mistaken :)

**Pull request merged**